src/main/java/org/sunbird/course/service/ExploreCourseServiceImpl.java

```diff
@@ -183,4 +185,135 @@ public SBApiResponse getExploreCourseListV2() {
 		}
 		return response;
 	}
+
+	@Override
+	public SBApiResponse upsertExploreCourse(Map<String, Object> requestObj) {
+		SBApiResponse response =
+				ProjectUtil.createDefaultResponse(Constants.API_EXPLORE_COURSE_UPDATE);
+		logger.info("ExploreCourseService::upsertExploreCourse:inside method");
+		Map<String, Object> masterData = (Map<String, Object>) requestObj.get(Constants.REQUEST);
+		String errMsg = validateUpsertRequest(masterData);
+		if (!StringUtils.isEmpty(errMsg)) {
+			response.getParams().setErrmsg(errMsg);
+			response.setResponseCode(HttpStatus.BAD_REQUEST);
+			return response;
+		}
+		try {
+			List<?> dataList = (List<?>) masterData.get(Constants.DATA);
+			Iterator<?> iterator = dataList.iterator();
+
+			while (iterator.hasNext()) {
+				Object item = iterator.next();
+				Map<?, ?> itemMap = (Map<?, ?>) item;
```

Comment:
variable item is not required... try to avoid creating local variables unless it is used repeatedly

this would be sufficient -> Map<?, ?> itemMap = (Map<?, ?>) iterator.next();

Reply:
made changes